File: DSA/DSA06028 SẮP XẾP CHỌN - LIỆT KÊ NGƯỢC.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    vector<int> A(n);
    for (auto &&i : A) cin >> i;
    vector<vector<int>> res;

    
    for (int i = 0; i < n-1; i++)
    {
        int mini = i;
        for (int j = i+1; j < n; j++)
        {
            if (A[j] < A[mini])
            {
                mini = j;
            }
        }
        swap(A[mini], A[i]);

        res.push_back(A);
    }
    
    for (int i = res.size()-1; i >= 0; i--)
    {
        
        cout << "Buoc " << i+1 << ": ";
        for (auto &&k : res[i]) cout << k << " ";
        cout << endl;
    }
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    submit();

    return 0;
}

File: DSA_python/DSA06028 SẮP XẾP CHỌN - LIỆT KÊ NGƯỢC.py
def submit():
    n = int(input())
    A = list(map(int, input().split()))
    res = []
    
    for i in range(n-1):
        mini = i
        for j in range(i+1, n):
            if A[j] < A[mini]:
                mini = j
        A[mini], A[i] = A[i], A[mini]

        res.append(A[:])

    for i in range(len(res)-1, -1, -1):
        print(f"Buoc {i+1}:", " ".join(map(str, res[i])))


submit()
